### ISSUE TITLE
beatCallback breaks when time signature numerator > denominator (e.g. 9/8)?

### ISSUE THREAD

**Issue by Esn024** [paulrosen/abcjs#797]
I tested beatCallback at various x/8 time signatures. It works fine from 2/8 up to 8/8, then breaks once the time signature becomes 9/8 - it suddenly runs at a much slower tempo than the actual playback of the music.  The audio playback of the song thus ends long before the final beatCallback does. Here's an example I used that shows the effect (try console logging something within the beatCallback while simultaneously playing the song as audio playback):
```js
const abc2 = `X:2
T:Title
C:Composer Name (yyyy)
M:9/8
Q:1/8=276
L:1/8
%%score (T1 T2)
V:T1           clef=treble  name="Left thumb"   snm="L"
V:T2           clef=treble  name="Right thumb"  snm="R"
K:Gm
%            End of header, start of tune body:
% 1
[V:T1]  zBzczdzga  | f6e2a      | (d2c2 d2)e2a | d4 c2z2a |
[V:T2]  GzAzBzeza  | d6c2a      | (B2A2 B2)c2a | B4 A2z2a |
% 5
[V:T1]  (B2c2 d2g2)a  | f8a        | d3c (d2fe)a  | H d6a    ||
[V:T2]       z8a      |     z8a    | B3A (B2c2)a  | H A6a    ||`;
```

This only seems to affect the beatCallback, as the eventCallback and sequenceCallback seem to work fine.

Do you know what might be going on?

I discovered that this was a problem just now when I attempted to set "Beats Per Measure" to more than "8" in the app I'm working on, [abcKalimba](https://abckalimba.herokuapp.com/) (however, I also tested it in simpler plain-JavaScript code and got the same bug, so I don't think it's React-specific).

**Comment by paulrosen** [comment]
I can't reproduce this. Can you post a minimal example? It seems to work for me with both the CursorControl and TimingCallbacks.

**Comment by paulrosen** [comment]
For abc5 I see what you are saying - in 6/8 there are 2 beats per measure and that works for the first part of the song. When it changes to 4/4 there are still 2 beats per measure but there should be 4.

For abc2, I still don't see the problem. In 9/8 there are 3 beats per measure and that looks like it tracks what is playing. you have 8 measures so that is 24 beats and I see the beat numbers go from 0 to 23 then go back to 0 when the tune starts over.

Do you want there to be 9 beats per measure?

**Comment by Esn024** [comment]
Indeed, I had assumed that the `L:` field ([unit note length](https://abcnotation.com/wiki/abc:standard:v2.1#lunit_note_length)), if it is present, would specify how many beats are intended. So, for 6/8, if `L:3/8`, there would be 2 beats a bar; if `L:2/8` there would be 3 beats a bar, and if `L:1/8`, there would be 6 beats a bar.

It seems that at the moment, for 9/8 ABCJS assumes that there are 3 beats per measure, while for 6/8, it assumes that there are 6 beats per measure.

I would like for the behaviour to be consistent and predictable, if possible.
